source/agora/consensus/validation/Block.d

```diff
@@ -116,6 +116,9 @@ public string isInvalidReason (in Block block, Engine engine, Height prev_height
     import std.conv;
     import std.range;
 
+    assert(prev_hash !is Hash.init);
+    assert(random_seed !is Hash.init);
+
     if (block.header.height == 0)
         return "Block: Genesis block should be validated using isGenesisBlockInvalidReason";
```

Comment:
considering that the GenesisBlock has prev_hash = Hash.init

```suggestion
	if (block.header.height == 0)
        return "Block: Genesis block should be validated using isGenesisBlockInvalidReason";
        
    assert(prev_hash !is Hash.init);
    assert(random_seed !is Hash.init);
```

we can also just delete the if

Reply:
Nice catch again, fixed